surface/model_openapiv3.go

```diff
@@ -439,3 +404,58 @@ func (b *OpenAPI3Builder) buildFromSchema(name string, schema *openapiv3.Schema)
 	log.Printf("Unimplemented: could not find field info for schema: %v", schema)
 	return nil
 }
+
+// buildFromOneOfAnyOfAndAllOf adds appropriate fields to the 'schemaType' given a new 'schemaOrRef'.
+func (b *OpenAPI3Builder) buildFromOneOfAnyOfAndAllOf(schemaOrRef *openapiv3.SchemaOrReference, schemaType *Type) {
+	// Related: https://github.com/googleapis/gnostic-grpc/issues/22
+	if schema := schemaOrRef.GetSchema(); schema != nil {
+		// Build a temporary type that has the required fields; add the fields to the current schema; remove the
+		// current type
+		fieldInfo := b.buildFromSchemaOrReference("ATemporaryTypeThatWillBeRemoved", schemaOrRef)
+		if t := findType(b.model.Types, "ATemporaryTypeThatWillBeRemoved"); t != nil {
+			for _, f := range t.Fields {
+				schemaType.Fields = append(schemaType.Fields, f)
+			}
+			b.removeType(t)
+		} else {
+			// schemaOrRef is some kind of primitive schema (e.g. of type string)
+			makeFieldAndAppendToType(fieldInfo, schemaType, "value")
+		}
```

Comment:
Style comment: could we invert the structuring of this `if`/`else` to reduce indentation? In the spirit of https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

```go
t := findType(b.model.Types, "ATemporaryTypeThatWillBeRemoved")
if t == nil {
    // schemaOrRef is some kind of primitive schema (e.g. of type string)
    makeFieldAndAppendToType(fieldInfo, schemaType, "value")
    return
}

schemaType.Fields = append(schemaType.Fields, t.Fields...)
b.removeType(t)
```

This puts the "primitive" scenario in the conditional flow with a terminating `return` and the "non-primitive" flow as the regular path of execution (in the scope of the `schema != nil` condition)

Feel free to ignore this, I just find it easier to read/cleaner.

Reply:
Done!

🙈Thanks Noah. I think we had that two or three times already. When I wrote that if-else statement I was like: "Yup, Noah won't approve this"

Reply:
Hahaha thanks for fixing it. I'm relentless on the if-else statements 😉